pkg/osquery/tables/pwpolicy/pwpolicy.go

```diff
@@ -0,0 +1,133 @@
+//+build darwin
+
+package pwpolicy
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	client       *osquery.ExtensionManagerClient
+	logger       log.Logger
+	tableName    string
+	execDataFunc func([]byte, ...dataflatten.FlattenOpts) ([]dataflatten.Row, error)
+}
+
+const PwpolicyPath = "/usr/bin/pwpolicy"
+const PwpolicyCmd = "getaccountpolicies"
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("fullkey"),
+		table.TextColumn("parent"),
+		table.TextColumn("key"),
+		table.TextColumn("value"),
+		table.TextColumn("query"),
+		table.TextColumn("username"),
+	}
+
+	t := &Table{
+		client:       client,
+		logger:       level.NewFilter(logger, level.AllowInfo()),
+		tableName:    "kolide_pwpolicy",
+		execDataFunc: dataflatten.Plist,
+	}
+
+	return table.NewPlugin(t.tableName, columns, t.generateExec)
+}
+
+func (t *Table) generateExec(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+	var username string
+
+	if q, ok := queryContext.Constraints["username"]; ok && len(q.Constraints) != 0 {
+		username = q.Constraints[0].Expression
+	}
```

Comment:
This will work. but I think there are a couple of issues.

First, it's a singleton -- it won't handle being passed multiple `username` columns. It should instead, be a fore loop.

Second, I recently got annoyed at needing the same `if q, ok ...` stuff, and wrote a helper for this. Check out `tablehelpers.GetConstraints` You can see an example of this over in tablehelpers.GetConstraints Note the `tablehelpers.WithDefaults("")` there.

Reply:
Fixed by copying similar pattern in ioreg